### PR TITLE
cursor: Split out cursor_button_press() and cursor_button_release()

### PR DESCRIPTION
**Pull request by jlindgren90**
The control flow in `cursor_button()` is a bit hard to follow, especially the `goto mousebindings` part. Looking at it closely, the logic for press events and release events is different enough that I think it makes sense to split out two smaller functions, `cursor_button_press()` and `cursor_button_release()`.

This includes and builds on #544, which should be merged first.

### CODE REVIEW

**Comment by Consolatis** [comment]
Code LGTM, will test later

**Comment by johanmalm** [comment]
Thank you! Looks so much better.

**Comment by Consolatis** [comment]
And testing didn't show any issues either. So nice to actually get a quick grasp on what is going on in those functions now :)
